content/blog/arduino-esp-now-lib/index.md

```diff
@@ -0,0 +1,456 @@
+---
+title: "Using ESP-NOW in Arduino"
+date: 2024-06-03T09:00:24-03:00
+tags: ["Arduino", "ESP-NOW", "Wireless", "Wi-Fi", "ESP32", "ESP32-S2", "ESP32-S3", "ESP32-C3", "ESP32-C6"]
+showAuthor: false
+authors:
+  - "lucas-vaz"
+---
+
+ESP-NOW is a connectionless Wi-Fi communication protocol developed by Espressif for its microcontrollers. It allows for efficient, low-power, and low-latency peer-to-peer communication. This article will guide you through the essentials of using the new ESP-NOW library included in Arduino 3.0.0, from basic setup to advanced features.
+
+## What is ESP-NOW?
+
+ESP-NOW enables direct communication between multiple devices without needing a router, ideal for applications requiring real-time data transfer with minimal delay. Key features include:
+- **Low Latency:** Real-time data transfer.
+- **Low Power Consumption:** Suitable for battery-powered devices.
+- **Peer-to-Peer Communication:** Direct device-to-device communication.
+- **Scalability:** Supports multiple peers in a network.
+- **Security:** Built-in encryption for secure data transfer.
+
+## Getting Started with ESP-NOW in Arduino
+
+To use ESP-NOW with Arduino, you need:
+- Arduino IDE installed on your computer;
+- Arduino Core for ESP32 installed version 3.0.0 or later.
+
+If you still haven't installed the Arduino Core for ESP32, you can do so by following the steps in the [official documentation](https://docs.espressif.com/projects/arduino-esp32/en/latest/getting_started.html).
+
+Note that this guide focuses on the new ESP-NOW library and assumes you are familiar with Arduino/C++ programming and the ESP32 platform.
+
+## Basic ESP-NOW Setup
+
+The ESP-NOW library in Arduino provides a set of classes and functions to simplify the setup and management of ESP-NOW communication. The library includes classes for handling peers, sending and receiving data, and managing the ESP-NOW network.
+
+The basic setup for ESP-NOW involves creating a custom class that inherits from the `ESP_NOW_Peer` class to handle the communication with a peer device. This class will define the send and receive callbacks and manage the ESP-NOW communication.
+
+After creating the custom class, you can create an instance of it passing the MAC address of the peer device, the Wi-Fi channel, the Wi-Fi interface, and the Local Master Key (LMK). Then, you can call the `ESP_NOW.begin()` and `add()` methods to initialize the ESP-NOW communication and register the peer device. Finally, you can use the `send()` method to send data to the peer device and the `onReceive()` method to handle received data.
+
+Be sure to check the [ESP-NOW Library Documentation](https://docs.espressif.com/projects/arduino-esp32/en/latest/api/espnow.html) for detailed information on the classes and methods available in the ESP-NOW library.
+
+Let's show the capabilities of the ESP-NOW library with some examples.
+
+### Two-Device Communication
```

Comment:
Do you think that some diagram or pictures could help people visualizing how ESP-NOW works?

Reply:
Added some pictures and diagrams